crates/wasmparser/src/readers/core/types.rs

```diff
@@ -691,6 +796,20 @@ pub struct StructType {
     pub fields: Box<[FieldType]>,
 }
 
+impl Matches for StructuralType {
+    fn matches<'a, F>(&self, other: &Self, type_at: &F) -> Result<bool>
+    where
+        F: Fn(u32) -> Result<&'a SubType>,
+    {
+        match (self, other) {
+            (StructuralType::Func(a), StructuralType::Func(b)) => a.matches(b, type_at),
+            (StructuralType::Array(a), StructuralType::Array(b)) => a.matches(b, type_at),
+            (StructuralType::Struct(a), StructuralType::Struct(b)) => a.matches(b, type_at),
+            _ => Ok(false),
```

Comment:
Like above I'd recommend removing catch-all cases like this, and instead here you can do something like:

```
(StructuralType::Func(a), StructuralType::Func(b)) => a.matches(b, type_at),
(StructuralType::Func(_), _) => false,
```

Reply:
Removed.